### PR TITLE
jenkins: Introduce jenkins-job-builder for jobs

### PR DESCRIPTION
**Pull request by toabctl**
This commit uses a yaml file to describe the openstack-cleanvm job.

### CODE REVIEW

**Comment by dirkmueller** [comment]
Love it, +100

**Comment by vuntz** [comment]
Nice. Do we have a way to make it clear in jenkins that the job should not be modified there, though?

**Comment by toabctl** [comment]
@vuntz: it's in the description. I already deployed the job.habe a look there.

**Comment by vuntz** [comment]
> @vuntz: it's in the description. I already deployed the job.habe a look there.

Ah indeed. Not super visible, though. I wonder if we can make jenkins somehow block direct configuration...

**Comment by toabctl** [comment]
> Ah indeed. Not super visible, though. I wonder if we can make jenkins somehow block direct configuration...

I don't know if it's possible to make a job only writable by jenkins-job-builder. Imo we should just use something like chef (or saltstack;)) to enforce the config...

**Comment by toabctl** [comment]
and fyi - the package is on the way to devel:languages:python (see https://build.opensuse.org/request/show/341684)

**Comment by aspiers** [comment]
Thanks, this is roughly 1,000,000 times nicer than XML plus a custom Perl hack ...

File: README.md [hunk 1]
```diff
@@ -68,3 +68,9 @@ jenkins/jenkinsnotify.pl syntax OK
 jenkins/openstack-unittest-testconfig.pl syntax OK
 jenkins/track-upstream-and-package.pl syntax OK
 ```
+
+# jenkins jobs
+There are manually maintained jobs and some jobs are now using `jenkins-job-builder` which defines jobs in yaml format.
```
Comment (aspiers) [inline README.md]: This should be a hyperlink to http://docs.openstack.org/infra/jenkins-job-builder/

Also nitpick: the line is too long ;-)

Comment (toabctl) [inline README.md]: @aspiers both done

**Comment by jdsn** [comment]
My proposal would be to retract the rights to from the team to modify the jobs we track that way (long term ideally all), this can be configured per job.

I'd like to add to this solution an auto-depoly service to not depend on $somedevelopers being present. So whenever there was a change to a jenkins job it gets deployed automatically. This could be done via Jenkins itself (via a job that monitors our repo) or via an event listener (the one that @toabctl also proposed) on a shared VM (like our trollolo-VM, where we run automation jobs that all developers should be able to access).

**Comment by jdsn** [comment]
oh, btw. I almost forgot:  :+1:

**Pull request merged**